app/views/govuk_component/related_items.raw.html.erb

```diff
@@ -1,7 +1,7 @@
 <% if local_assigns[:sections] && !sections.blank? %>
   <aside class="govuk-related-items" data-module="track-click" role="complementary">
     <% sections.each_with_index do |section, section_index| %>
-      <h2
+      <h2 data-track-count="sidebarRelatedItemSection"
```

Comment:
Maybe you've considered this and disregarded it so sorry, but instead of the JavaScript checking the amount of elements on the page could we potentially do something like.

```erb
<aside
  class="govuk-related-items"
  data-module="track-click"
  data-track-total-sections="<%= sections.length %>" 
  role="complementary"
>
```

One con of this is it could drop out of sync with what users see but interested to hear your thoughts since it would make your logic simpler I think.